### PR TITLE
add: pull request preview.

### PR DESCRIPTION
**Pull request by cissero**
pull request preview:
add the possibility to add an action.
choose an action in the list and confirm your choice with the '+' button.

### CODE REVIEW

File: dahu/core/app/scripts/collections/actions.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ define([
                 default:
                     return new ActionModel(attrs, options);
             }
+        },
+        getAvailableActions: function(){
```
Comment (barraq) [inline dahu/core/app/scripts/collections/actions.js:33]: simply return ["move", "appear", "disappear"]

File: dahu/core/app/scripts/controller/layout.js [hunk 2]
```diff
@@ -139,7 +139,6 @@ define([
             }));
 
             this._registerToScreenEditorEvents();
-
```
Comment (barraq) [inline dahu/core/app/scripts/controller/layout.js:142]: ??? :beer:

File: dahu/core/app/scripts/models/actions/appear.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ define([
     var AppearModel = ActionModel.extend({
         defaults: function() {
             return _.extend({}, ActionModel.prototype.defaults(), {
-                type: 'appear'
```
Comment (barraq) [inline dahu/core/app/scripts/models/actions/appear.js:17]: ? why ?

File: dahu/core/app/scripts/models/actions/disappear.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ define([
     var DisappearModel = ActionModel.extend({
         defaults: function() {
             return _.extend({}, ActionModel.prototype.defaults(), {
-                type: 'disappear'
```
Comment (barraq) [inline dahu/core/app/scripts/models/actions/disappear.js:17]: ? why ?

File: dahu/core/app/scripts/models/actions/move.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ define([
     var MoveModel = ActionModel.extend({
         defaults: function() {
             return _.extend({}, ActionModel.prototype.defaults(), {
-                type: 'move',
```
Comment (barraq) [inline dahu/core/app/scripts/models/actions/move.js:16]: ? why ?

File: dahu/core/app/scripts/controller/layout.js (hunk 2 above)
Comment (cissero) [inline dahu/core/app/scripts/controller/layout.js:142]: :) oki for the  beer.

File: dahu/core/app/scripts/views/workspace/actions.js [hunk 6]
```diff
@@ -61,6 +58,36 @@ define([
                 }
             }*/
         },
+        templateHelpers: function () {
+           return{
+               ActionsAvailable: this.collection.getAvailableActions(),
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: respect camel case

File: dahu/core/app/scripts/views/workspace/actions.js [hunk 7]
```diff
@@ -61,6 +58,36 @@ define([
...
+                    break;
+                }
+                case "disappear":{
+                    this.collection.add(new DisappearModel());
+                    break;
+                }
+                default:
+                    this.collection.add(new AppearModel());
+            }
+            this.$childViewContainer[0].scrollTop=this.$childViewContainer[0].scrollHeight;
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: ? why ?

File: dahu/core/app/scripts/views/workspace/actions.js [hunk 8]
```diff
@@ -61,6 +58,36 @@ define([
                 }
             }*/
         },
+        templateHelpers: function () {
+           return{
+               ActionsAvailable: this.collection.getAvailableActions(),
+           }
+        },
+        events: {
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: use trigger instead

File: dahu/core/app/scripts/views/workspace/actions.js [hunk 9]
```diff
@@ -61,6 +58,36 @@ define([
...
+          'click .buttonAdd': 'clickedButton',
+          'change .addAction': 'addActionChoice'
+        },
+
+        clickedButton: function() {
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: rename it to onCreateAction

File: dahu/core/app/scripts/views/workspace/actions.js [hunk 10]
```diff
@@ -61,6 +58,36 @@ define([
...
+                case "appear":{
+                    this.collection.add(new AppearModel());
+                    break;
+                }
+                case "disappear":{
+                    this.collection.add(new DisappearModel());
+                    break;
+                }
+                default:
+                    this.collection.add(new AppearModel());
```
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: a value must be provided.... default must through an exception.

File: dahu/core/app/styles/main.scss [hunk 11]
```diff
@@ -179,4 +179,34 @@ html, body {
...
+            z-index: 10;
+            .addAction{
+                left:0;
+               
+            }
+            .buttonAdd {
+                right:0;
+            }
+        }
+        .listActions {
```
Comment (barraq) [inline dahu/core/app/styles/main.scss]: use actionsList

File: dahu/core/app/scripts/views/workspace/actions.js (hunk 6 above)
Comment (cissero) [inline dahu/core/app/scripts/views/workspace/actions.js]: ok

File: dahu/core/app/styles/main.scss [hunk 12]
```diff
@@ -179,4 +179,34 @@ html, body {
     .action:last-child::after {
 	border-bottom: none;
     }
+    
+     #ActionToolbar {
```
Comment (barraq) [inline dahu/core/app/styles/main.scss]: use camel case

File: dahu/core/app/scripts/templates/views/workspace/actions.html [hunk 13]
```diff
@@ -1,3 +1,11 @@
-<!--	
-	Empty on purpose
--->
\ No newline at end of file
+<div class="listActions">
+<div class="emptyActionDiv"/>
```
Comment (barraq) [inline dahu/core/app/scripts/templates/views/workspace/actions.html:2]: why ?

File: dahu/core/app/scripts/views/workspace/actions.js (hunk 7 above)
Comment (cissero) [inline dahu/core/app/scripts/views/workspace/actions.js]: it is to focus on the action added recently.The action is added at the end of the actionsList so without that user must scroll to see and edit the added  action.With this line the list is automatically scrolled when we add an action.

File: dahu/core/app/scripts/templates/views/workspace/actions.html [hunk 14]
```diff
@@ -1,3 +1,11 @@
-<!--	
-	Empty on purpose
--->
\ No newline at end of file
+<div class="listActions">
```
Comment (barraq) [inline dahu/core/app/scripts/templates/views/workspace/actions.html]: use #actionsList instead

File: dahu/core/app/scripts/templates/views/workspace/actions.html [hunk 15]
```diff
@@ -1,3 +1,11 @@
...
+<div class="emptyActionDiv"/>
+</div>
+<div id="ActionToolbar">
```
Comment (barraq) [inline dahu/core/app/scripts/templates/views/workspace/actions.html]: use camel case, rename it to actionsToolbar

File: dahu/core/app/scripts/views/workspace/actions.js (hunk 9 above)
Comment (cissero) [inline dahu/core/app/scripts/views/workspace/actions.js]: ok

File: dahu/core/app/scripts/views/workspace/actions.js (hunk 7 above)
Comment (barraq) [inline dahu/core/app/scripts/views/workspace/actions.js]: I mean the line 89 ?

File: dahu/core/app/scripts/views/workspace/actions.js (hunk 10 above)
Comment (cissero) [inline dahu/core/app/scripts/views/workspace/actions.js]: ok

File: dahu/core/app/styles/main.scss (hunk 12 above)
Comment (cissero) [inline dahu/core/app/styles/main.scss]: ok

File: dahu/core/app/styles/main.scss (hunk 11 above)
Comment (cissero) [inline dahu/core/app/styles/main.scss]: The div which contains listActions and ActionToolbar is named ActionsList...

**Pull request closed without merging**